### ISSUE TITLE
Watch mode does not pick up code changes

### ISSUE THREAD

**Issue by mkampmey** [vuejs/language-tools#2090]
Today I updated vue-tsc from 0.40.13 to 1.0.9 and I noticed that watch mode ( `vue-tsc --noEmit --watch`) does not work anymore. So I have to restart vue-tsc every time when I fix a code line.

**Comment by cannap** [comment]
yes same here i dont know about the other versions but i have the same problem

**Comment by nick-w-m** [comment]
Same here. Using 1.0.9 and it does react when a file is saved

**Comment by nick-w-m** [comment]
Repro might exist on typescript <4.5 in package.json
When i bumped to typescript 4.5+ i got watch to work (aswell as ts errors in <template> blocks, which did not work on <4.5 either)

**Comment by cannap** [comment]
> 

```
{
  "name": "xxxxxxxi",
  "version": "0.0.0",
  "files": [
    "dist",
    "typings"
  ],
 ....
  "typings": "./types/index.d.ts",
  "scripts": {
    "dev": "vite",
    "build": "run-p type-check build-only",
    "build:lib": "vite build",
    "preview": "vite preview",
    "watch:lib": "vite build --watch",
    "build-only": "vite build && vue-tsc --declaration --emitDeclarationOnly",
    "type-check": "vue-tsc --noEmit",
    "create-types": "vue-tsc --declaration --emitDeclarationOnly",
    "lint": "eslint . --ext .vue,.js,.jsx,.cjs,.mjs,.ts,.tsx,.cts,.mts --fix --ignore-path .gitignore",
    "story:dev": "histoire dev",
    "story:build": "histoire build",
    "story:preview": "histoire preview"
  },
  "type": "module",
  "peerDependencies": {
    "vue": "^3.2.41"
  },
  "devDependencies": {
    "@histoire/plugin-vue": "^0.11.6",
    "@rushstack/eslint-patch": "^1.1.4",
    "@types/node": "^16.11.68",
    "@vitejs/plugin-vue": "^3.1.2",
    "@vue/eslint-config-prettier": "^7.0.0",
    "@vue/eslint-config-typescript": "^11.0.0",
    "@vue/tsconfig": "^0.1.3",
    "cross-env": "^7.0.3",
    "esbuild": "^0.15.13",
    "eslint": "^8.22.0",
    "eslint-plugin-vue": "^9.3.0",
    "histoire": "^0.11.6",
    "npm-run-all": "^4.1.5",
    "prettier": "^2.7.1",
    "storybook": "^7.0.0-alpha.47",
    "typescript": "~4.7.4",
    "unplugin-vue-components": "^0.22.9",
    "vite": "^3.1.8",
    "vite-plugin-dts": "^1.6.6", // Not in usage right now 
    "vue-tsc": "^1.0.9"
  }
}
```

i have over 4.5 but still not watching

**Comment by johnsoncodehk** [comment]
Can you reproduce this problem in https://github.com/johnsoncodehk/volar-starter?

**Comment by littleboarx** [comment]
@johnsoncodehk any progress on this problem?

**Comment by johnsoncodehk** [comment]
@littleboarx It's weird, it only happens with npm, but yarn and pnpm work fine. I think it has something to do with some external dependency used by the volar modules, perhaps referencing the wrong version when using npm.

**Comment by mkampmey** [comment]
For me it works again, also with npm. I am using vue-tsc@1.0.24 and typescript@4.9.4.

**Comment by johnsoncodehk** [comment]
Fixed by #2580
